core/src/avm1/globals/color_transform.rs

```diff
@@ -0,0 +1,299 @@
+) -> Result<(), Error<'gc>> {
+    object.set("redMultiplier", color_transform.r_mult.into(), avm, context)?;
+    object.set(
+        "greenMultiplier",
+        color_transform.g_mult.into(),
+        avm,
+        context,
+    )?;
+    object.set(
+        "blueMultiplier",
+        color_transform.b_mult.into(),
+        avm,
+        context,
+    )?;
+    object.set(
+        "alphaMultiplier",
+        color_transform.a_mult.into(),
+        avm,
+        context,
+    )?;
+    object.set("redOffset", color_transform.r_add.into(), avm, context)?;
+    object.set("greenOffset", color_transform.g_add.into(), avm, context)?;
+    object.set("blueOffset", color_transform.b_add.into(), avm, context)?;
+    object.set("alphaOffset", color_transform.a_add.into(), avm, context)?;
+    Ok(())
+}
+
+pub fn object_to_color_transform<'gc>(
+    object: Object<'gc>,
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+) -> Result<ColorTransform, Error<'gc>> {
+    let red_multiplier = object
+        .get("redMultiplier", avm, context)?
+        .coerce_to_f64(avm, context)? as f32;
+    let green_multiplier = object
+        .get("greenMultiplier", avm, context)?
+        .coerce_to_f64(avm, context)? as f32;
+    let blue_multiplier = object
+        .get("blueMultiplier", avm, context)?
+        .coerce_to_f64(avm, context)? as f32;
+    let alpha_multiplier = object
+        .get("alphaMultiplier", avm, context)?
+        .coerce_to_f64(avm, context)? as f32;
+    let red_offset = object
+        .get("redOffset", avm, context)?
+        .coerce_to_f64(avm, context)? as f32;
+    let green_offset = object
+        .get("greenOffset", avm, context)?
+        .coerce_to_f64(avm, context)? as f32;
+    let blue_offset = object
+        .get("blueOffset", avm, context)?
+        .coerce_to_f64(avm, context)? as f32;
+    let alpha_offset = object
+        .get("alphaOffset", avm, context)?
+        .coerce_to_f64(avm, context)? as f32;
+
+    Ok(ColorTransform {
+        r_mult: red_multiplier,
+        g_mult: green_multiplier,
+        b_mult: blue_multiplier,
+        a_mult: alpha_multiplier,
+        r_add: red_offset,
+        g_add: green_offset,
+        b_add: blue_offset,
+        a_add: alpha_offset,
+    })
+}
+
+pub fn get_rgb<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error<'gc>> {
+    let ct = object_to_color_transform(this, avm, context)?;
+    let rgb = ((ct.r_add as u32) << 16) | ((ct.g_add as u32) << 8) | (ct.b_add as u32);
+    Ok(Value::Number(rgb.into()).into())
+}
+
+pub fn set_rgb<'gc>(
+    avm: &mut Avm1<'gc>,
+    context: &mut UpdateContext<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error<'gc>> {
+    let new_rgb = args
+        .get(0)
+        .unwrap_or(&Value::Undefined)
+        .coerce_to_f64(avm, context)? as u64;
```

Comment:
This should be `coerce_to_u32` because it takes care of the necessary wrapping behavior. If `f64` is huge, Rust (recently) defines float-to-int casts to saturate, which won't produce the correct value. Try:

```
var ct = new flash.geom.ColorTransform();
var n = 36893488147419103232;
ct.rgb = n;
trace(ct);
```

Flash:
`(redMultiplier=0, greenMultiplier=0, blueMultiplier=0, alphaMultiplier=1, redOffset=0, greenOffset=0, blueOffset=0, alphaOffset=0)`
Ruffle:
`(redMultiplier=0, greenMultiplier=0, blueMultiplier=0, alphaMultiplier=1, redOffset=255, greenOffset=255, blueOffset=255, alphaOffset=0)
`